### PR TITLE
fix #1153 preventing multiple threads working on same bulk file

### LINKED ISSUE #1153: Bulk Import:  In Progress Check
At present Accumulo does not check for a bulk import in progress before initiating a new one.  If a bulk import is in progress and another one is started it's possible for the second one to error out if the files it needs are already processed by the first one.  Tablet.java currently has a check in place to see if the bulk import in progress is attempting to process files already successfully bulk imported:

https://github.com/apache/accumulo/blob/c56eadabf913f981fa10a51f9f874b4f6287351a/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java#L2433 

Adding a check to Tablet.java for bulk imports actively in progress would be of great benefit.  Perhaps this could be added in Tablet.java near the check or part of the check for bulk imports already completed.

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 1]
```diff
@@ -2437,6 +2442,16 @@ public void importMapFiles(long tid, Map<FileRef,MapFileInfo> fileMap, boolean s
           }
         }
       }
+
+      Iterator<FileRef> fiter = fileMap.keySet().iterator();
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: I bet there's a removeIf construct in Java 8 you can use for the master branch.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 2]
```diff
@@ -2471,6 +2489,11 @@ public void importMapFiles(long tid, Map<FileRef,MapFileInfo> fileMap, boolean s
         if (writesInProgress == 0)
           this.notifyAll();
 
+        if (!bulkImporting.removeAll(fileMap.keySet())) {
+          throw new AssertionError(
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: :+1: for the sanity check with the AssertionError. Good message, too.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 3]
```diff
@@ -255,6 +255,11 @@ public long getDataSourceDeletions() {
 
   private final ConfigurationObserver configObserver;
 
+  // Files that are currently in the process of bulk importing. Access to this is protected by the
+  // tablet lock.
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: Good comment about the lock. This answered a question I had.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java (hunk 1 above)
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: When I first wrote this code I used multiple Java 8 map methods.  Then I realized I couldn't.

This particular code could have been implement using `fileMap.keySet().removeAll(bulkImporting)`.  However then there would be no logging.  I thought the logging was important, so wrote it this way which was the most efficient way I could think of writing it.

**Pull request merged**